osgpkitools/incommon_request.py

```diff
@@ -0,0 +1,453 @@
+        default=None
+    )
+    parser.add_option(
+        '-a',
+        '--altname',
+        action='append',
+        dest='alt_names',
+        help='Specify an alternative hostname for CSR (FQDN). May be used more than ' + \
+             'once and if specified, -f/--hostfile will be ignored',
+        metavar='HOSTNAME',
+        default=[]
+    )
+    parser.add_option(
+        '-u',
+        '--username',
+        action='store',
+        dest='login',
+        help='Provide the InCommon username (login).',
+        metavar='LOGIN',
+        default=[]
+    )
+    parser.add_option(
+        '-T',
+        '--test',
+        action='store_true',
+        dest='test',
+        help='Test connection to InCommon API. Useful to test authentication credentials',
+        default=False
+    )
+    parser.add_option(
+        '',
+        '--debug',
+        dest='debug',
+        help="Write debug output to stdout",
+        action='store_true',
+        default=False
+    )
+    
+    parser.add_option_group(group)
+    (args, values) = parser.parse_args()
+
+    if args.debug:
+        # this sets the root debug level
+        logging.getLogger().setLevel(logging.DEBUG)
+        logger.debug('Debug mode enabled')
+    
+    if not args.login:
+        raise InsufficientArgumentException("InsufficientArgumentException: " + \
+                                            "Please provide the InCommon username (login)\n")
+        
+    if not args.test:    
+        if not args.hostname:
+            if args.hostfile is None:
+                raise InsufficientArgumentException("InsufficientArgumentException: " + \
+                                                    "Please provide hostname(-H) or file name containing hosts(-f)\n")
+            else:
+                    hostfile = args.hostfile
+        else:
+            hostname = args.hostname
+        
+    if not args.test:
+        if not args.hostname:
+            if os.path.exists(hostfile):
+                pass
+            else:
+                raise FileNotFoundException(hostfile, 'Error: could not locate the hostfile')
+            
+
+    arguments = dict()
+
+    if vars().has_key('args'):
+        arguments.update({'args': args})
+    if vars().has_key('values'):
+        arguments.update({'values': values})
+    if vars().has_key('hostname'):
+        arguments.update({'hostname': hostname})
+    
+    arguments.update({'alt_names': args.alt_names})
+    arguments.update({'test': args.test})
+    arguments.update({'login': args.login})
+    
+    if args.usercert and args.userprivkey:
+        usercert, userkey = find_user_cred(args.usercert, args.userprivkey)
+        arguments.update({'usercert': usercert})
+        arguments.update({'userprivkey': userkey})
+
+    arguments.update({'certdir': args.write_directory})
+    
+    if vars().has_key('hostfile'):
+        arguments.update({'hostfile': hostfile})
+    
+    return arguments
+
+def buildHeaders(config):
```

Comment:
Can we pep8 function and variable names?

https://www.python.org/dev/peps/pep-0008/#function-and-variable-names